account-kit/core/src/types.ts

```diff
@@ -62,23 +62,68 @@ export type AlchemyAccountsConfig = {
     sessionLength: number;
   };
 };
-
 // [!region CreateCorConfigProps]
-export type Connection = {
+export type ViemConnection = {
   transport: AlchemyTransportConfig;
   chain: Chain;
   policyId?: string | string[];
 };
 
+export type Web3Chain = Pick<Chain, "name" | "nativeCurrency"> & { id: string };
+
+export type Web3Connection = {
+  connection: SolanaWeb3Connection;
+  chain: Web3Chain;
+  policyId?: string;
+};
+/**
+ * Checks if the connection is a viem/ transport connection
+ *
+ * @param {Connection} connection - The connection to check
+ * @returns {connection is ViemConnect} true if the connection is a Viem connection, false otherwise
+ */
+export function isViemConnection(
+  connection: Connection
+): connection is ViemConnection {
+  return "transport" in connection;
```

Comment:
could we add a `type: "eth" | "solana"` to the connection object to make this checking a little less implicit?

calling it `eth` and `solana` also is a bit clearer cuz web3 and viem requires quite a bit of context to understand what they imply